Test/PerformanceCollector/FunctionalTests/Serialization/TelemetryExtensions.cs

```diff
@@ -40,6 +41,30 @@ public static class TelemetryExtensions
             return result;
         }
 
+        internal static MonitoringDataPoint[] ReceiveItems(
+          this QuickPulseHttpListenerObservable listener,
+          int count,
+          int timeOut)
+        {
+            if (null == listener)
+            {
+                throw new ArgumentNullException("listener");
+            }
+
+            var result = listener
+                .TakeUntil(DateTimeOffset.UtcNow.AddMilliseconds(timeOut))
+                .Take(count)
+                .ToEnumerable()
```

Comment:
Isn't it already IEnumerable (or derived)?  I've always been able to call .ToArray() directly on LINQ query results.